plugins/statistics.koplugin/main.lua

```diff
@@ -105,6 +105,16 @@ function ReaderStatistics:init()
         return -- disable in PIC documents
     end
 
+    if self.document and self.document.file and self.document.file:match("^/tmp/") then
+        -- entirely disable for temporary files, including disabling event handlers
+        for n,v in pairs(ReaderStatistics) do
+            if n:match("^on") and type(v) == "function" then
+                self[n] = function() end
+            end
+        end
```

Comment:
This is a bit ugly:
1) forcing this /tmp/ check in here, may be on some platform you'll need to use something other than /tmp/.
2) and stripping down ReaderStatistic on all its event handlers is a bit of a hack - I'd rather you keep that done in your plugin.

Dunno if there could be some other cleaner way do do that, like we do for pic documents:
https://github.com/koreader/koreader/blob/030f4a15d4a2b13d3264968bf10d7b13b0c3ec97/plugins/statistics.koplugin/main.lua#L103-L106
but you'll need to hook into document or Reader before your plugin is activated :/ I can't think of how right now.

We have ReaderStatistics:isEnabled() which uses self.settings.is_enabled.
So, even with your stripping of ReaderStatistics, users will see statistics are enabled in the menu, and may get confused.
I think if you want some support for that in core (vs. hacking it from your plugin), it would have to be cleaner and more generic. May be some idea will come to us.

Reply:
Plus, IIRC, the event handlers are already setup to handle the plugin being disabled that way, so I don't really see the interest in murdering them? (Unless there's like a billion triggers in your plugin?).

(Speaking of murdering event handlers: `Plugin.onEvent = nil` works just fine,; and is even better than a NOP).

Reply:
Just returning early in init() as for PIC didn't work, the event handlers were still called leading to errors because self.settings was nil.

Reply:
You could also try : `self.ui.statistics = nil`, no idea how things would behave (as :init() plugs a few functions in ReaderFooter, BookStatusWidget, Screensaver...)